### PR TITLE
[TT-7345] Adding the possibility to create historical and future data in Pump's Demo Mode

### PR DESCRIPTION
**Pull request by mativm02**
## Description

Now users can choose between creating historical or future data when using Pump's Demo Mode. It will generate historical data by default.
## Related Issue

https://tyktech.atlassian.net/browse/TT-7345?atlOrigin=eyJpIjoiZDBlN2I0YTlmYzc0NGIwZWFhNTI5YTMyMzg2NzAyODUiLCJwIjoiaiJ9
## Motivation and Context

It allows users to create more realistic data. Historical data makes a lot more sense than future data, that's the reason of the change.
## How This Has Been Tested

Unit tests + executing demo mode using Mongo Pump

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

File: analytics/demo/demo.go [hunk 1]
```diff
@@ -149,43 +149,60 @@ func country() string {
...
-
-				set = append(set, r)
+	// If we are generating future data, we want to start at the current date and create data for the next X days
+	if demoFutureData {
+		for d := 0; d < days; d++ {
+			for h := 0; h < 24; h++ {
+				WriteDemoData(start, d, h, recordsPerHour, orgID, trackPath, writer)
 			}
+			count++
+			fmt.Printf("Finished %d of %d\n", count, days)
```
Comment (sredxny) [inline analytics/demo/demo.go:163]: can we use a logger instead?

File: README.md [hunk 2]
```diff
@@ -1301,3 +1301,4 @@ You can run Tyk Pump in demo mode, which will generate fake analytics data and s
 - `--demo-days=<DAYS>` - Sets the number of days of demo data to generate. Defaults to 30.
 - `--demo-records-per-hour=<RECORDS_PER_HOUR>` - Sets the number of records to generate per hour. The default value is a random number between 300 and 500.
 - `--demo-track-path` - Enables tracking of the request path in the demo data. Defaults to false (disabled). Note that setting `track_all_paths` to `true` in your Pump configuration will override this option.
+- `--demo-future-data` - By default, the demo data is generated for the past X days (configured in `demo-days` flag). This option will generate data for the next X days. Defaults to false (disabled).
```
Comment (sredxny) [inline README.md:1304]: Can I generate past data and future data at the same time?

**Review by sredxny — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (mativm02) [reply README.md:1304]: No, you can choose between one option or the other one, but not both at the same time

File: analytics/demo/demo.go (hunk 1 above)
Comment (mativm02) [reply analytics/demo/demo.go]: For sure! it was added

**Review by sredxny — APPROVED** [review]
lgtm

**Pull request merged**